internal/build/promote.go

```diff
@@ -29,12 +31,21 @@ func Promote(opts Opts) error {
 	for _, crossRelease := range cat.Releases.Items {
 		release := crossRelease.Releases[0]
 		if release.Spec.Project == opts.Project {
+
 			// Find version node
 			versionNode, err := yml.FindNode(release.File.Tree, "spec.version")
 			if err != nil {
 				return fmt.Errorf("release %s has no version property: %w", release.Name, err)
 			}
 
+			// Check the version format in source Release
```

Comment:
I think that this would be clearer if it wasn't done inside of the loop. What we need to check is that version is valid or not for the environment. I think you could do this before entering the loop:

```go
isPr := func() bool {
	version := "v" + opts.Version
	return semver.Prerelease(version)+semver.Build(version) != ""
}()

if isPr && !cat.Environments[0].Spec.Promotion.FromPullRequests {
	return fmt.Errorf("cannot promote release with non-standard version to %s environment", opts.Environment)
}
```